services/src/main/java/org/exoplatform/addons/gamification/storage/dao/RuleDAO.java

```diff
@@ -203,6 +201,9 @@ private <T> void addQueryFilterParameters(RuleFilter filter, TypedQuery<T> query
     if (CollectionUtils.isNotEmpty(filter.getSpaceIds())) {
       query.setParameter("ids", filter.getSpaceIds());
     }
+    if (filter.getFilterType() != null && !FilterType.ALL.equals(filter.getFilterType())) {
+      query.setParameter("date", Utils.parseSimpleDate( Utils.toSimpleDateFormat(new Date(System.currentTimeMillis()))));
```

Comment:
why do you need to make such transformation on date object to pass it as parameter ?
This should be sufficient : 
```suggestion
      query.setParameter("date", Calendar.getInstance().getTime());
```

Reply:
done!